src/content/reference/rules/never-pass-around-hooks-as-regular-values.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Never pass around hooks as regular values
+---
+
+<Intro>
+Hooks should only be called inside of components. Never pass it around as a regular value.
+</Intro>
+
+---
+
+Hooks allow you to augment a component with React features. They should always be called as a function, and never passed around as a regular value. This enables _local reasoning_, or the ability for developers to understand everything a component can do by looking at that component in isolation.
+
+```js {2}
+function ChatInput() {
+  const useDataWithLogging = withLogging(useData); // ❌
+  const data = useDataWithLogging();
+}
+```
+
+```js {2}
+function ChatInput() {
+  const data = useDataWithLogging(); // ✅
+}
+```
+
+Hooks should be immutable and not be mutated. Instead of mutating a hook dynamically, create a static version of the hook with the desired functionality.
+
+```js
+function useDataWithLogging() {
+  // ... Logic should go in here
+}
+```
+
+Hooks should also not be dynamically used: for example, instead of doing dependency injection in a component:
+
+```js {2}
+function ChatInput() {
+  return <Button useData={useDataWithLogging} /> // ❌
```

Comment:
I'm not aware that we talked about this rule before publicly. Especially because it doesn't violate the original rules of hooks as far as I can tell.

I checked our codebase for potential violations by just grepping for `use*` as props with `use[A-Z]\w+=`.

Out of the 8000+ components we have (including the ones without any props), only 2 accepted hooks as props. So the rule wouldn't prohibit a popular pattern in our case. 

The two cases where very trivial where you could've just passed the return value of the hook instead of the hook itself:

```diff
-function Child({ useSettings }) {
-   const settings = useSettings
+function Child({ settings }) {
 }


 function Parent() {
+ const settings = useSettings()
-  return <Child useSettings={useSettings} />
+  return <Child settings={settings} />
 }
```

Though this is a slightly different change now because there may be cases where you wanted  to spread hook calls across different components for perf and moving all calls into a single component may be bad since it makes a single render call too expensive.

There's another instance in our codebase where we pass hooks as value but I'm all for breaking that up since nobody understands that piece of code.